src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/CommunityAPIOffenderService.kt

```diff
@@ -86,4 +104,24 @@ class CommunityAPIOffenderService(
       .block()
       ?.staffIdentifier
   }
+
+  fun getResponsibleOfficer(crn: String): ResponsibleOfficer {
+    val offenderManagersPath = UriComponentsBuilder.fromPath(offenderManagersLocation)
+      .buildAndExpand(crn)
+      .toString()
+
+    // technically this endpoint can return more than one offender manager that
+    // is a responsible officer. in practice this should not happen. if it does
+    // we just take the first one we come across.
+    return communityApiClient.get(offenderManagersPath)
+      .retrieve()
+      .bodyToFlux(OffenderManagerResponse::class.java)
+      .filter { it.isResponsibleOfficer == true }
+      .map { ResponsibleOfficer(parseName(it.staff?.forenames, it.staff?.surname), it.staff?.email) }
+      .blockFirst()
+  }
+
+  private fun parseName(forenames: String?, surname: String?): String? {
```

Comment:
Do you have any tests for this? 

(I also think it would be helpful to contain the unit tests in the same commit)

Reply:
currently no - i can add. i'll move the code to a public utility method called 'mungeNames' or something.

on the latter i agree - but the prospect of the rebase and potential manual merges put me off spending the time to do it. how important do you think this is in terms of the commit history ?

Reply:
The last comment was just as a general rule as it makes it easier to review - I was not expecting you to make that change in this PR.